source/material_model/rheology/diffusion_creep.cc

```diff
@@ -39,52 +39,91 @@ namespace aspect
+            creep_parameters.activation_energy = MaterialModel::MaterialUtilities::phase_average_value(phase_function_values, n_phases_per_composition,
+                                                 activation_energies_diffusion, composition);
+            creep_parameters.activation_volume = MaterialModel::MaterialUtilities::phase_average_value(phase_function_values, n_phases_per_composition,
+                                                 activation_volumes_diffusion, composition);
+            creep_parameters.stress_exponent = MaterialModel::MaterialUtilities::phase_average_value(phase_function_values, n_phases_per_composition,
+                                               stress_exponents_diffusion, composition);
+            creep_parameters.grain_size_exponent = MaterialModel::MaterialUtilities::phase_average_value(phase_function_values, n_phases_per_composition,
+                                                   grain_size_exponents_diffusion, composition);
           }
+        return creep_parameters;
+      }
+
+
+      template <int dim>
+      double
+      DiffusionCreep<dim>::compute_viscosity (const double pressure,
+                                              const double temperature,
+                                              const unsigned int composition,
+                                              const std::vector<double> &phase_function_values,
+                                              const std::vector<unsigned int> &n_phases_per_composition) const
+      {
+        const DiffusionCreepParameters p = compute_creep_parameters(composition,
+                                                                    phase_function_values,
+                                                                    n_phases_per_composition);
+
         // Power law creep equation
         //    viscosity = 0.5 * A^(-1/n) * d^(m/n) * exp((E + P*V)/(nRT))
         // A: prefactor,
         // d: grain size, m: grain size exponent, E: activation energy, P: pressure,
         // V; activation volume, R: gas constant, T: temperature.
-        const double viscosity_diffusion = 0.5 / prefactors_diffusion_composition *
-                                           std::exp((activation_energies_diffusion_composition +
-                                                     pressure*activation_volumes_diffusion_composition)/
+        const double viscosity_diffusion = 0.5 / p.prefactor *
+                                           std::exp((p.activation_energy +
+                                                     pressure*p.activation_volume)/
                                                     (constants::gas_constant*temperature)) *
-                                           std::pow(grain_size, grain_size_exponents_diffusion_composition);
+                                           std::pow(grain_size, p.grain_size_exponent);
 
         return viscosity_diffusion;
       }
 
 
```

Comment:
add another empty line (3 lines between functions), same below this function

Reply:
Done (also in other places here and in dislocation_creep.cc